modules/xfd.js

```diff
@@ -0,0 +1,2052 @@
+        // Ensure items with User talk or no namespace prefix both end
+        // up at user talkspace as expected, but retain the
+        // prefix-less username for addToLog
+        var params = this.params;
+        var def = $.Deferred();
+        var notifyTitle = mw.Title.newFromText(notifyTarget, 3); // 3: user talk
+        var targetNS = notifyTitle.getNamespaceId();
+        var usernameOrTarget = notifyTitle.getRelativeText(3);
+        statusElement = statusElement || new Morebits.status('Notifying initial contributor (' + usernameOrTarget + ')');
+        var notifyPageTitle = notifyTitle.toText();
+        if (targetNS === 3) {
+            // Disallow warning yourself
+            if (usernameOrTarget === mw.config.get('wgUserName')) {
+                params.initialContrib = null; // disable initial contributor logging in userspace log
+                statusElement.warn('You (' + usernameOrTarget + ') created this page; skipping user notification');
+                return def.resolve();
+            }
+        }
+        var usertalkpage = new Morebits.wiki.page(notifyPageTitle, statusElement);
+        usertalkpage.setAppendText('\n\n' + this.getNotifyText());
+        usertalkpage.setEditSummary(this.getNotifyEditSummary());
+        usertalkpage.setChangeTags(Twinkle.changeTags);
+        usertalkpage.setCreateOption('recreate');
+        // Different pref for RfD target notifications: XXX: handle this better!
+        if (params.venue === 'rfd' && targetNS !== 3) {
+            usertalkpage.setWatchlist(Twinkle.getPref('xfdWatchRelated'));
+        }
+        else {
+            usertalkpage.setWatchlist(Twinkle.getPref('xfdWatchUser'));
+        }
+        usertalkpage.setFollowRedirect(true, false);
+        usertalkpage.append(def.resolve, function onNotifyError() {
+            // if user could not be notified, null this out for correct userspace logging
+            params.initialContrib = null;
+            def.resolve();
+        });
+        return def;
+    };
+    // Overridden for all venues except FFD and RFD
+    XfdMode.prototype.getNotifyText = function () {
+        return "{{subst:" + this.params.venue + " notice|1=" + Morebits.pageNameNorm + "}} ~~~~";
+    };
+    // Not overridden for any venue
+    XfdMode.prototype.getNotifyEditSummary = function () {
+        return 'Notification: [[' + this.params.discussionpage + '|listing]] of [[:' +
+            Morebits.pageNameNorm + ']] at [[WP:' + this.getFieldsetLabel() + ']].';
+    };
+    XfdMode.prototype.notifyCreator = function () {
+        if (!this.params.notifycreator) {
+            this.params.intialContrib = null;
+            return $.Deferred().resolve();
+        }
+        return this.notifyTalkPage(this.params.initialContrib);
+    };
+    // Should be called after notifyTalkPage() which may unset this.params.intialContrib
+    XfdMode.prototype.addToLog = function () {
+        var params = this.params;
```

Comment:
params is only needed once, suggest using this.params.venue instead of needing a temporary variable